src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2841,14 +2861,17 @@ ManageWorkerPool(WorkerPool *workerPool)
 
 		if (execution->failed)
 		{
+			const char *errHint =
+				execution->localExecutionSupported ?
+				"Consider enabling local execution using SET "
+				"citus.enable_local_execution TO true;" :
+				"Consider using a higher value for max_connections";
+
 			ereport(ERROR, (errcode(ERRCODE_CONNECTION_FAILURE),
-							errmsg(
-								"could not establish any connections to the node %s:%d "
-								"when local execution is also disabled.",
-								workerPool->nodeName,
-								workerPool->nodePort),
-							errhint("Enable local execution via SET "
-									"citus.enable_local_execution TO true;")));
+							errmsg("could not establish any connections to the "
```

Comment:
`due to high number connections` sounds vague. It depends on the max_connections and *high* depends on that.
Maybe: `ERROR:The total number of connections on the server is more than max_connections(%s) `, max_connections  
`HINT1`: This command supports local execution. Consider enabliing ..rest is same ....
`HINT2`: Keep same